index.js

```diff
@@ -0,0 +1,97 @@
+require('dotenv').config();
+const express = require('express');
+const cors = require('cors');
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+
+const db = require('./database/dbConfig');
+const server = express();
+
+const generateToken = (user) => {
+
+    const payload = {
+        subject: user.userId,
+        username: user.username,
+        roles: ['admin']
+    }
+
+    const secret = process.env.JWT_SECRET;
+    const options = {
+        expiresIn: '1h'
+    }
+
+    return jwt.sign(payload, secret, options)
+}
+
+const protected = (req, res, next) => {
+    const token = req.headers.authentication;
```

Comment:
With way how it is setup here it will only work with Your client. The general common way is to search for `Authorization` in the headers. Your way follows the `authentication` which will not work with most of the apps outside Your React client.